### PR TITLE
[SW-2485] Deprecate Apache Spark 2.1

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: core/src/main/scala/ai/h2o/sparkling/H2OContext.scala [hunk 1]
```diff
@@ -488,6 +493,11 @@ object H2OContext extends Logging {
           s" but your Spark is of version ${sc.version}. Please make sure to use correct Sparkling Water for your" +
           s" Spark and re-run the application.")
     }
+    if (isSparkVersionDeprecated) {
+      logWarning(
+        "Apache Spark 2.1.x is deprecated and " +
```
Comment (honzasterba) [inline core/src/main/scala/ai/h2o/sparkling/H2OContext.scala:498]: I would make this parametrized by the current version `SparkSessionUtils.active.version` as well

File: core/src/main/scala/ai/h2o/sparkling/H2OContext.scala [hunk 2]
```diff
@@ -467,9 +467,14 @@ object H2OContext extends Logging {
     getOrCreate(Option(instantiatedContext.get()).map(_.getConf).getOrElse(new H2OConf()))
   }
 
+  private def isSparkVersionDeprecated(): Boolean = {
+    SparkSessionUtils.active.version.startsWith("2.1")
```
Comment (satai) [inline core/src/main/scala/ai/h2o/sparkling/H2OContext.scala:471]: What about spark 2.10? ;-)
(I know, it doesn't exist but I just warn about this pattern.)

Comment (mn-mikke) [reply core/src/main/scala/ai/h2o/sparkling/H2OContext.scala:471]: ok, I will improve that

File: core/src/main/scala/ai/h2o/sparkling/H2OContext.scala (hunk 1 above)
Comment (mn-mikke) [reply core/src/main/scala/ai/h2o/sparkling/H2OContext.scala:498]: will do

**Review by honzasterba — APPROVED** [review]

**Pull request merged**